### ISSUE TITLE
HTTPS Redirect Loop when using ForceSSL on CentOS7 with Apache

### ISSUE THREAD

**Issue by KyferEz** [phpmyadmin/phpmyadmin#11883]
CentOS 7 x64 fully updated. EPEL repositories. Everything working perfectly except phpMyAdmin. I cannot get ForceSSL to work properly. It results in a redirect loop. I don't know if this is an issue with my server config or phpMyAdmin.

I have been troubleshooting by adding some logging into "function detectHttps()". What I have found is that it ALWAYS falls thru to this line of code in detectHttps:
```
            } else {
                $url['scheme'] = 'http';
```
phpMyAdmin version:
```
Version information: 4.4.15.2 (up to date)
```
HTTPd -v
```
Server version: Apache/2.4.6 (CentOS)
Server built:   Nov 19 2015 21:43:13
```

Figure you may want to see my phpinfo (identifying info replaced with generic info):
```
PHP Logo
PHP Version 5.4.16


System	Linux s01 3.10.0-327.4.4.el7.x86_64 #1 SMP Tue Jan 5 16:07:00 UTC 2016 x86_64
Build Date	Jun 23 2015 21:18:22
Server API	Apache 2.0 Handler
Virtual Directory Support	disabled
Configuration File (php.ini) Path	/etc
Loaded Configuration File	/etc/php.ini
Scan this dir for additional .ini files	/etc/php.d
Additional .ini files parsed	/etc/php.d/bcmath.ini, /etc/php.d/curl.ini, /etc/php.d/dom.ini, /etc/php.d/fileinfo.ini, /etc/php.d/gd.ini, /etc/php.d/json.ini, /etc/php.d/mbstring.ini, /etc/php.d/mcrypt.ini, /etc/php.d/mysqlnd.ini, /etc/php.d/mysqlnd_mysql.ini, /etc/php.d/mysqlnd_mysqli.ini, /etc/php.d/pdo.ini, /etc/php.d/pdo_mysqlnd.ini, /etc/php.d/pdo_sqlite.ini, /etc/php.d/phar.ini, /etc/php.d/posix.ini, /etc/php.d/sqlite3.ini, /etc/php.d/sysvmsg.ini, /etc/php.d/sysvsem.ini, /etc/php.d/sysvshm.ini, /etc/php.d/tidy.ini, /etc/php.d/wddx.ini, /etc/php.d/xmlreader.ini, /etc/php.d/xmlwriter.ini, /etc/php.d/xsl.ini, /etc/php.d/zip.ini
PHP API	20100412
PHP Extension	20100525
Zend Extension	220100525
Zend Extension Build	API220100525,NTS
PHP Extension Build	API20100525,NTS
Debug Build	no
Thread Safety	disabled
Zend Signal Handling	disabled
Zend Memory Manager	enabled
Zend Multibyte Support	provided by mbstring
IPv6 Support	enabled
DTrace Support	disabled
Registered PHP Streams	https, ftps, compress.zlib, compress.bzip2, php, file, glob, data, http, ftp, phar, zip
Registered Stream Socket Transports	tcp, udp, unix, udg, ssl, sslv3, sslv2, tls
Registered Stream Filters	zlib.*, bzip2.*, convert.iconv.*, string.rot13, string.toupper, string.tolower, string.strip_tags, convert.*, consumed, dechunk, mcrypt.*, mdecrypt.*

Zend logo This program makes use of the Zend Scripting Language Engine:
Zend Engine v2.4.0, Copyright (c) 1998-2013 Zend Technologies

PHP Credits

Configuration

apache2handler

Apache Version	Apache/2.4.6 (CentOS) OpenSSL/1.0.1e-fips PHP/5.4.16
Apache API Version	20120211
Server Administrator	root@localhost
Hostname:Port	123.123.123.123:0
User/Group	apache(48)/48
Max Requests	Per Child: 0 - Keep Alive: on - Max Per Connection: 100
Timeouts	Connection: 60 - Keep-Alive: 5
Virtual Server	Yes
Server Root	/etc/httpd
Loaded Modules	core mod_so http_core mod_access_compat mod_actions mod_alias mod_allowmethods mod_auth_basic mod_auth_digest mod_authn_anon mod_authn_core mod_authn_dbd mod_authn_dbm mod_authn_file mod_authn_socache mod_authz_core mod_authz_dbd mod_authz_dbm mod_authz_groupfile mod_authz_host mod_authz_owner mod_authz_user mod_autoindex mod_cache mod_cache_disk mod_data mod_dbd mod_deflate mod_dir mod_dumpio mod_echo mod_env mod_expires mod_ext_filter mod_filter mod_headers mod_include mod_info mod_log_config mod_logio mod_mime_magic mod_mime mod_negotiation mod_remoteip mod_reqtimeout mod_rewrite mod_setenvif mod_slotmem_plain mod_slotmem_shm mod_socache_dbm mod_socache_memcache mod_socache_shmcb mod_status mod_substitute mod_suexec mod_unique_id mod_unixd mod_userdir mod_version mod_vhost_alias mod_dav mod_dav_fs mod_dav_lock mod_lua prefork mod_proxy mod_lbmethod_bybusyness mod_lbmethod_byrequests mod_lbmethod_bytraffic mod_lbmethod_heartbeat mod_proxy_ajp mod_proxy_balancer mod_proxy_connect mod_proxy_express mod_proxy_fcgi mod_proxy_fdpass mod_proxy_ftp mod_proxy_http mod_proxy_scgi mod_proxy_wstunnel mod_ssl mod_systemd mod_cgi mod_php5

Directive	Local Value	Master Value
engine	1	1
last_modified	0	0
xbithack	0	0

Apache Environment

Variable	Value
UNIQUE_ID	sdfghfvsrzPldUkxO5yA-QAAAAo
HTTP_HOST	123.123.123.123
HTTP_CONNECTION	keep-alive
HTTP_CACHE_CONTROL	max-age=0
HTTP_ACCEPT	text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
HTTP_UPGRADE_INSECURE_REQUESTS	1
HTTP_USER_AGENT	Mozilla/5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.82 Safari/537.36
HTTP_ACCEPT_ENCODING	gzip, deflate, sdch
HTTP_ACCEPT_LANGUAGE	en-US,en;q=0.8
HTTP_COOKIE	testing=1; sid=ac430136e1e63a636308f3960eb05833; PHPSESSID=asdfghtle0bf42sgf8326485f3
PATH	/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin
SERVER_SIGNATURE	no value
SERVER_SOFTWARE	Apache/2.4.6 (CentOS) OpenSSL/1.0.1e-fips PHP/5.4.16
SERVER_NAME	123.123.123.123
SERVER_ADDR	172.20.5.75
SERVER_PORT	80
REMOTE_ADDR	11.11.11.11
DOCUMENT_ROOT	/var/www/html/example.com/html
REQUEST_SCHEME	http
CONTEXT_PREFIX	no value
CONTEXT_DOCUMENT_ROOT	/var/www/html/example.com/html
SERVER_ADMIN	root@localhost
SCRIPT_FILENAME	/var/www/html/example.com/html/phpinfo.php
REMOTE_PORT	52028
GATEWAY_INTERFACE	CGI/1.1
SERVER_PROTOCOL	HTTP/1.1
REQUEST_METHOD	GET
QUERY_STRING	no value
REQUEST_URI	/phpinfo.php
SCRIPT_NAME	/phpinfo.php

HTTP Headers Information

HTTP Request Headers
HTTP Request	GET /phpinfo.php HTTP/1.1
Host	123.123.123.123
Connection	keep-alive
Cache-Control	max-age=0
Accept	text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
Upgrade-Insecure-Requests	1
User-Agent	Mozilla/5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.82 Safari/537.36
Accept-Encoding	gzip, deflate, sdch
Accept-Language	en-US,en;q=0.8
Cookie	testing=1; sid=ac430136e1e63a636308f3960eb05833; PHPSESSID=asdfghtle0bf42sgf8326485f3
HTTP Response Headers
X-Powered-By	PHP/5.4.16

bcmath

BCMath support	enabled

Directive	Local Value	Master Value
bcmath.scale	0	0

bz2

BZip2 Support	Enabled
Stream Wrapper support	compress.bzip2://
Stream Filter support	bzip2.decompress, bzip2.compress
BZip2 Version	1.0.6, 6-Sept-2010

calendar

Calendar support	enabled

Core

PHP Version	5.4.16

Directive	Local Value	Master Value
allow_url_fopen	On	On
allow_url_include	Off	Off
always_populate_raw_post_data	Off	Off
arg_separator.input	&	&
arg_separator.output	&	&
asp_tags	Off	Off
auto_append_file	no value	no value
auto_globals_jit	On	On
auto_prepend_file	no value	no value
browscap	no value	no value
default_charset	no value	no value
default_mimetype	text/html	text/html
disable_classes	no value	no value
disable_functions	no value	no value
display_errors	Off	Off
display_startup_errors	Off	Off
doc_root	no value	no value
docref_ext	no value	no value
docref_root	no value	no value
enable_dl	Off	Off
enable_post_data_reading	On	On
error_append_string	no value	no value
error_log	no value	no value
error_prepend_string	no value	no value
error_reporting	22527	22527
exit_on_timeout	Off	Off
expose_php	On	On
extension_dir	/usr/lib64/php/modules	/usr/lib64/php/modules
file_uploads	On	On
highlight.comment	#FF8000	#FF8000
highlight.default	#0000BB	#0000BB
highlight.html	#000000	#000000
highlight.keyword	#007700	#007700
highlight.string	#DD0000	#DD0000
html_errors	On	On
ignore_repeated_errors	Off	Off
ignore_repeated_source	Off	Off
ignore_user_abort	Off	Off
implicit_flush	Off	Off
include_path	.:/usr/share/pear:/usr/share/php	.:/usr/share/pear:/usr/share/php
log_errors	On	On
log_errors_max_len	1024	1024
mail.add_x_header	On	On
mail.force_extra_parameters	no value	no value
mail.log	no value	no value
max_execution_time	30	30
max_file_uploads	20	20
max_input_nesting_level	64	64
max_input_time	60	60
max_input_vars	1000	1000
memory_limit	128M	128M
open_basedir	no value	no value
output_buffering	4096	4096
output_handler	no value	no value
post_max_size	8M	8M
precision	14	14
realpath_cache_size	16K	16K
realpath_cache_ttl	120	120
register_argc_argv	Off	Off
report_memleaks	On	On
report_zend_debug	On	On
request_order	GP	GP
sendmail_from	no value	no value
sendmail_path	/usr/sbin/sendmail -t -i	/usr/sbin/sendmail -t -i
serialize_precision	17	17
short_open_tag	Off	Off
SMTP	localhost	localhost
smtp_port	25	25
sql.safe_mode	Off	Off
track_errors	Off	Off
unserialize_callback_func	no value	no value
upload_max_filesize	2M	2M
upload_tmp_dir	no value	no value
user_dir	no value	no value
user_ini.cache_ttl	300	300
user_ini.filename	.user.ini	.user.ini
variables_order	GPCS	GPCS
xmlrpc_error_number	0	0
xmlrpc_errors	Off	Off
zend.detect_unicode	On	On
zend.enable_gc	On	On
zend.multibyte	Off	Off
zend.script_encoding	no value	no value

ctype

ctype functions	enabled

curl

cURL support	enabled
cURL Information	7.29.0
Age	3
Features
AsynchDNS	Yes
Debug	No
GSS-Negotiate	Yes
IDN	Yes
IPv6	Yes
Largefile	Yes
NTLM	Yes
SPNEGO	No
SSL	Yes
SSPI	No
krb4	No
libz	Yes
CharConv	No
Protocols	dict, file, ftp, ftps, gopher, http, https, imap, imaps, ldap, ldaps, pop3, pop3s, rtsp, scp, sftp, smtp, smtps, telnet, tftp
Host	x86_64-redhat-linux-gnu
SSL Version	NSS/3.19.1 Basic ECC
ZLib Version	1.2.7
libSSH Version	libssh2/1.4.3

date

date/time support	enabled
"Olson" Timezone Database Version	0.system
Timezone Database	internal
Default timezone	UTC

Directive	Local Value	Master Value
date.default_latitude	31.7667	31.7667
date.default_longitude	35.2333	35.2333
date.sunrise_zenith	90.583333	90.583333
date.sunset_zenith	90.583333	90.583333
date.timezone	no value	no value

dom

DOM/XML	enabled
DOM/XML API Version	20031129
libxml Version	2.9.1
HTML Support	enabled
XPath Support	enabled
XPointer Support	enabled
Schema Support	enabled
RelaxNG Support	enabled

ereg

Regex Library	Bundled library enabled

exif

EXIF Support	enabled
EXIF Version	1.4 $Id$
Supported EXIF Version	0220
Supported filetypes	JPEG,TIFF

Directive	Local Value	Master Value
exif.decode_jis_intel	JIS	JIS
exif.decode_jis_motorola	JIS	JIS
exif.decode_unicode_intel	UCS-2LE	UCS-2LE
exif.decode_unicode_motorola	UCS-2BE	UCS-2BE
exif.encode_jis	no value	no value
exif.encode_unicode	ISO-8859-15	ISO-8859-15

fileinfo

fileinfo support	enabled
version	1.0.5

filter

Input Validation and Filtering	enabled
Revision	$Id: 2aa8dd57d9c0c655cd45e6e5872bb95fa5ad76cf $

Directive	Local Value	Master Value
filter.default	unsafe_raw	unsafe_raw
filter.default_flags	no value	no value

ftp

FTP support	enabled

gd

GD Support	enabled
GD Version	bundled (2.1.0 compatible)
FreeType Support	enabled
FreeType Linkage	with freetype
FreeType Version	2.4.11
T1Lib Support	enabled
GIF Read Support	enabled
GIF Create Support	enabled
JPEG Support	enabled
libJPEG Version	6b
PNG Support	enabled
libPNG Version	1.5.13
WBMP Support	enabled
XPM Support	enabled
libXpm Version	30411
XBM Support	enabled

Directive	Local Value	Master Value
gd.jpeg_ignore_warning	0	0

gettext

GetText Support	enabled

gmp

gmp support	enabled
GMP version	6.0.0

hash

hash support	enabled
Hashing Engines	md2 md4 md5 sha1 sha224 sha256 sha384 sha512 ripemd128 ripemd160 ripemd256 ripemd320 whirlpool tiger128,3 tiger160,3 tiger192,3 tiger128,4 tiger160,4 tiger192,4 snefru snefru256 gost adler32 crc32 crc32b fnv132 fnv164 joaat haval128,3 haval160,3 haval192,3 haval224,3 haval256,3 haval128,4 haval160,4 haval192,4 haval224,4 haval256,4 haval128,5 haval160,5 haval192,5 haval224,5 haval256,5

iconv

iconv support	enabled
iconv implementation	glibc
iconv library version	2.17

Directive	Local Value	Master Value
iconv.input_encoding	ISO-8859-1	ISO-8859-1
iconv.internal_encoding	ISO-8859-1	ISO-8859-1
iconv.output_encoding	ISO-8859-1	ISO-8859-1

json

json support	enabled
json version	1.2.1

libxml

libXML support	active
libXML Compiled Version	2.9.1
libXML Loaded Version	20901
libXML streams	enabled

mbstring

Multibyte Support	enabled
Multibyte string engine	libmbfl
HTTP input encoding translation	disabled
libmbfl version	1.3.2

mbstring extension makes use of "streamable kanji code filter and converter", which is distributed under the GNU Lesser General Public License version 2.1.

Multibyte (japanese) regex support	enabled
Multibyte regex (oniguruma) backtrack check	On
Multibyte regex (oniguruma) version	4.7.1

Directive	Local Value	Master Value
mbstring.detect_order	no value	no value
mbstring.encoding_translation	Off	Off
mbstring.func_overload	0	0
mbstring.http_input	pass	pass
mbstring.http_output	pass	pass
mbstring.http_output_conv_mimetypes	^(text/|application/xhtml\+xml)	^(text/|application/xhtml\+xml)
mbstring.internal_encoding	no value	no value
mbstring.language	neutral	neutral
mbstring.strict_detection	Off	Off
mbstring.substitute_character	no value	no value

mcrypt

mcrypt support	enabled
mcrypt_filter support	enabled
Version	2.5.8
Api No	20021217
Supported ciphers	cast-128 gost rijndael-128 twofish arcfour cast-256 loki97 rijndael-192 saferplus wake blowfish-compat des rijndael-256 serpent xtea blowfish enigma rc2 tripledes
Supported modes	cbc cfb ctr ecb ncfb nofb ofb stream

Directive	Local Value	Master Value
mcrypt.algorithms_dir	no value	no value
mcrypt.modes_dir	no value	no value

mhash

MHASH support	Enabled
MHASH API Version	Emulated Support

mysql

MySQL Support	enabled
Active Persistent Links	0
Active Links	0
Client API version	mysqlnd 5.0.10 - 20111026 - $Id: e707c415db32080b3752b232487a435ee0372157 $

Directive	Local Value	Master Value
mysql.allow_local_infile	On	On
mysql.allow_persistent	On	On
mysql.connect_timeout	60	60
mysql.default_host	no value	no value
mysql.default_password	no value	no value
mysql.default_port	no value	no value
mysql.default_socket	/var/lib/mysql/mysql.sock	/var/lib/mysql/mysql.sock
mysql.default_user	no value	no value
mysql.max_links	Unlimited	Unlimited
mysql.max_persistent	Unlimited	Unlimited
mysql.trace_mode	Off	Off

mysqli

MysqlI Support	enabled
Client API library version	mysqlnd 5.0.10 - 20111026 - $Id: e707c415db32080b3752b232487a435ee0372157 $
Active Persistent Links	0
Inactive Persistent Links	0
Active Links	0

Directive	Local Value	Master Value
mysqli.allow_local_infile	On	On
mysqli.allow_persistent	On	On
mysqli.default_host	no value	no value
mysqli.default_port	3306	3306
mysqli.default_pw	no value	no value
mysqli.default_socket	/var/lib/mysql/mysql.sock	/var/lib/mysql/mysql.sock
mysqli.default_user	no value	no value
mysqli.max_links	Unlimited	Unlimited
mysqli.max_persistent	Unlimited	Unlimited
mysqli.reconnect	Off	Off

mysqlnd

mysqlnd	enabled
Version	mysqlnd 5.0.10 - 20111026 - $Id: e707c415db32080b3752b232487a435ee0372157 $
Compression	supported
SSL	supported
Command buffer size	4096
Read buffer size	32768
Read timeout	31536000
Collecting statistics	Yes
Collecting memory statistics	No
Tracing	n/a
Loaded plugins	mysqlnd,example,debug_trace,auth_plugin_mysql_native_password,auth_plugin_mysql_clear_password
API Extensions	mysql,mysqli,pdo_mysql

mysqlnd statistics	
bytes_sent	0
bytes_received	0
packets_sent	0
packets_received	0
protocol_overhead_in	0
protocol_overhead_out	0
bytes_received_ok_packet	0
bytes_received_eof_packet	0
bytes_received_rset_header_packet	0
bytes_received_rset_field_meta_packet	0
bytes_received_rset_row_packet	0
bytes_received_prepare_response_packet	0
bytes_received_change_user_packet	0
packets_sent_command	0
packets_received_ok	0
packets_received_eof	0
packets_received_rset_header	0
packets_received_rset_field_meta	0
packets_received_rset_row	0
packets_received_prepare_response	0
packets_received_change_user	0
result_set_queries	0
non_result_set_queries	0
no_index_used	0
bad_index_used	0
slow_queries	0
buffered_sets	0
unbuffered_sets	0
ps_buffered_sets	0
ps_unbuffered_sets	0
flushed_normal_sets	0
flushed_ps_sets	0
ps_prepared_never_executed	0
ps_prepared_once_executed	0
rows_fetched_from_server_normal	0
rows_fetched_from_server_ps	0
rows_buffered_from_client_normal	0
rows_buffered_from_client_ps	0
rows_fetched_from_client_normal_buffered	0
rows_fetched_from_client_normal_unbuffered	0
rows_fetched_from_client_ps_buffered	0
rows_fetched_from_client_ps_unbuffered	0
rows_fetched_from_client_ps_cursor	0
rows_affected_normal	0
rows_affected_ps	0
rows_skipped_normal	0
rows_skipped_ps	0
copy_on_write_saved	0
copy_on_write_performed	0
command_buffer_too_small	0
connect_success	0
connect_failure	0
connection_reused	0
reconnect	0
pconnect_success	0
active_connections	0
active_persistent_connections	0
explicit_close	0
implicit_close	0
disconnect_close	0
in_middle_of_command_close	0
explicit_free_result	0
implicit_free_result	0
explicit_stmt_close	0
implicit_stmt_close	0
mem_emalloc_count	0
mem_emalloc_amount	0
mem_ecalloc_count	0
mem_ecalloc_amount	0
mem_erealloc_count	0
mem_erealloc_amount	0
mem_efree_count	0
mem_efree_amount	0
mem_malloc_count	0
mem_malloc_amount	0
mem_calloc_count	0
mem_calloc_amount	0
mem_realloc_count	0
mem_realloc_amount	0
mem_free_count	0
mem_free_amount	0
mem_estrndup_count	0
mem_strndup_count	0
mem_estndup_count	0
mem_strdup_count	0
proto_text_fetched_null	0
proto_text_fetched_bit	0
proto_text_fetched_tinyint	0
proto_text_fetched_short	0
proto_text_fetched_int24	0
proto_text_fetched_int	0
proto_text_fetched_bigint	0
proto_text_fetched_decimal	0
proto_text_fetched_float	0
proto_text_fetched_double	0
proto_text_fetched_date	0
proto_text_fetched_year	0
proto_text_fetched_time	0
proto_text_fetched_datetime	0
proto_text_fetched_timestamp	0
proto_text_fetched_string	0
proto_text_fetched_blob	0
proto_text_fetched_enum	0
proto_text_fetched_set	0
proto_text_fetched_geometry	0
proto_text_fetched_other	0
proto_binary_fetched_null	0
proto_binary_fetched_bit	0
proto_binary_fetched_tinyint	0
proto_binary_fetched_short	0
proto_binary_fetched_int24	0
proto_binary_fetched_int	0
proto_binary_fetched_bigint	0
proto_binary_fetched_decimal	0
proto_binary_fetched_float	0
proto_binary_fetched_double	0
proto_binary_fetched_date	0
proto_binary_fetched_year	0
proto_binary_fetched_time	0
proto_binary_fetched_datetime	0
proto_binary_fetched_timestamp	0
proto_binary_fetched_string	0
proto_binary_fetched_blob	0
proto_binary_fetched_enum	0
proto_binary_fetched_set	0
proto_binary_fetched_geometry	0
proto_binary_fetched_other	0
init_command_executed_count	0
init_command_failed_count	0
com_quit	0
com_init_db	0
com_query	0
com_field_list	0
com_create_db	0
com_drop_db	0
com_refresh	0
com_shutdown	0
com_statistics	0
com_process_info	0
com_connect	0
com_process_kill	0
com_debug	0
com_ping	0
com_time	0
com_delayed_insert	0
com_change_user	0
com_binlog_dump	0
com_table_dump	0
com_connect_out	0
com_register_slave	0
com_stmt_prepare	0
com_stmt_execute	0
com_stmt_send_long_data	0
com_stmt_close	0
com_stmt_reset	0
com_stmt_set_option	0
com_stmt_fetch	0
com_deamon	0
bytes_received_real_data_normal	0
bytes_received_real_data_ps	0

example statistics	
stat1	0
stat2	0

openssl

OpenSSL support	enabled
OpenSSL Library Version	OpenSSL 1.0.1e-fips 11 Feb 2013
OpenSSL Header Version	OpenSSL 1.0.1e-fips 11 Feb 2013

pcre

PCRE (Perl Compatible Regular Expressions) Support	enabled
PCRE Library Version	8.32 2012-11-30

Directive	Local Value	Master Value
pcre.backtrack_limit	1000000	1000000
pcre.recursion_limit	100000	100000

PDO

PDO support	enabled
PDO drivers	mysql, sqlite

pdo_mysql

PDO Driver for MySQL	enabled
Client API version	mysqlnd 5.0.10 - 20111026 - $Id: e707c415db32080b3752b232487a435ee0372157 $

Directive	Local Value	Master Value
pdo_mysql.default_socket	/var/lib/mysql/mysql.sock	/var/lib/mysql/mysql.sock

pdo_sqlite

PDO Driver for SQLite 3.x	enabled
SQLite Library	3.7.17

Phar

Phar: PHP Archive support	enabled
Phar EXT version	2.0.1
Phar API version	1.1.1
SVN revision	$Id: c5042cc34acebcc0926625b57dff03deebbe6472 $
Phar-based phar archives	enabled
Tar-based phar archives	enabled
ZIP-based phar archives	enabled
gzip compression	enabled
bzip2 compression	enabled
Native OpenSSL support	enabled

Phar based on pear/PHP_Archive, original concept by Davey Shafik.
Phar fully realized by Gregory Beaver and Marcus Boerger.
Portions of tar implementation Copyright (c) 2003-2009 Tim Kientzle.

Directive	Local Value	Master Value
phar.cache_list	no value	no value
phar.readonly	On	On
phar.require_hash	On	On

posix

Revision	$Id: 32db6705f5b617967a546be3114e178a4138c1ca $

Reflection

Reflection	enabled
Version	$Id: 6c4d8062369898a397e4b128348042f5c01b4427 $

session

Session Support	enabled
Registered save handlers	files user
Registered serializer handlers	php php_binary wddx

Directive	Local Value	Master Value
session.auto_start	Off	Off
session.cache_expire	180	180
session.cache_limiter	nocache	nocache
session.cookie_domain	no value	no value
session.cookie_httponly	Off	Off
session.cookie_lifetime	0	0
session.cookie_path	/	/
session.cookie_secure	Off	Off
session.entropy_file	/dev/urandom	/dev/urandom
session.entropy_length	32	32
session.gc_divisor	1000	1000
session.gc_maxlifetime	1440	1440
session.gc_probability	1	1
session.hash_bits_per_character	5	5
session.hash_function	0	0
session.name	PHPSESSID	PHPSESSID
session.referer_check	no value	no value
session.save_handler	files	files
session.save_path	/var/lib/php/session	no value
session.serialize_handler	php	php
session.upload_progress.cleanup	On	On
session.upload_progress.enabled	On	On
session.upload_progress.freq	1%	1%
session.upload_progress.min_freq	1	1
session.upload_progress.name	PHP_SESSION_UPLOAD_PROGRESS	PHP_SESSION_UPLOAD_PROGRESS
session.upload_progress.prefix	upload_progress_	upload_progress_
session.use_cookies	On	On
session.use_only_cookies	On	On
session.use_trans_sid	0	0

shmop

shmop support	enabled

SimpleXML

Simplexml support	enabled
Revision	$Id: 692516840b2d7d6e7aedb0bedded1f53b764a99f $
Schema support	enabled

sockets

Sockets Support	enabled

SPL

SPL support	enabled
Interfaces	Countable, OuterIterator, RecursiveIterator, SeekableIterator, SplObserver, SplSubject
Classes	AppendIterator, ArrayIterator, ArrayObject, BadFunctionCallException, BadMethodCallException, CachingIterator, CallbackFilterIterator, DirectoryIterator, DomainException, EmptyIterator, FilesystemIterator, FilterIterator, GlobIterator, InfiniteIterator, InvalidArgumentException, IteratorIterator, LengthException, LimitIterator, LogicException, MultipleIterator, NoRewindIterator, OutOfBoundsException, OutOfRangeException, OverflowException, ParentIterator, RangeException, RecursiveArrayIterator, RecursiveCachingIterator, RecursiveCallbackFilterIterator, RecursiveDirectoryIterator, RecursiveFilterIterator, RecursiveIteratorIterator, RecursiveRegexIterator, RecursiveTreeIterator, RegexIterator, RuntimeException, SplDoublyLinkedList, SplFileInfo, SplFileObject, SplFixedArray, SplHeap, SplMinHeap, SplMaxHeap, SplObjectStorage, SplPriorityQueue, SplQueue, SplStack, SplTempFileObject, UnderflowException, UnexpectedValueException

sqlite3

SQLite3 support	enabled
SQLite3 module version	0.7
SQLite Library	3.7.17

Directive	Local Value	Master Value
sqlite3.extension_dir	no value	no value

standard

Dynamic Library Support	enabled
Path to sendmail	/usr/sbin/sendmail -t -i

Directive	Local Value	Master Value
assert.active	1	1
assert.bail	0	0
assert.callback	no value	no value
assert.quiet_eval	0	0
assert.warning	1	1
auto_detect_line_endings	0	0
default_socket_timeout	60	60
from	no value	no value
url_rewriter.tags	a=href,area=href,frame=src,input=src,form=fakeentry	a=href,area=href,frame=src,input=src,form=fakeentry
user_agent	no value	no value

sysvmsg

sysvmsg support	enabled
Revision	$Id: e268047b695fe9a43cdf415c429729252347becd $

tidy

Tidy support	enabled
libTidy Release	25 March 2009
Extension Version	2.0 ($Id$)

Directive	Local Value	Master Value
tidy.clean_output	no value	no value
tidy.default_config	no value	no value

tokenizer

Tokenizer Support	enabled

wddx

WDDX Support	enabled
WDDX Session Serializer	enabled

xml

XML Support	active
XML Namespace Support	active
libxml2 Version	2.9.1

xmlreader

XMLReader	enabled

xmlwriter

XMLWriter	enabled

xsl

XSL	enabled
libxslt Version	1.1.28
libxslt compiled against libxml Version	2.9.1
EXSLT	enabled
libexslt Version	1.1.28

zip

Zip	enabled
Extension Version	$Id: 0c033d4e4613d577409950ed7bf8da4b68286d15 $
Zip version	1.11.0
Compiled against libzip version	0.10.1

zlib

ZLib Support	enabled
Stream Wrapper	compress.zlib://
Stream Filter	zlib.inflate, zlib.deflate
Compiled Version	1.2.7
Linked Version	1.2.7

Directive	Local Value	Master Value
zlib.output_compression	Off	Off
zlib.output_compression_level	-1	-1
zlib.output_handler	no value	no value

Additional Modules

Module Name
sysvsem
sysvshm

Environment

Variable	Value
LANG	C
PATH	/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin
NOTIFY_SOCKET	/run/systemd/notify

PHP Variables

Variable	Value
_COOKIE["testing"]	1
_COOKIE["sid"]	ac430136e1e63a636308f3960eb05833
_COOKIE["PHPSESSID"]	fmno1ctle0bf42sgf8326485f3
_SERVER["UNIQUE_ID"]	VqT90fvsrzPldUkxO5yA-QAAAAo
_SERVER["HTTP_HOST"]	123.123.123.123
_SERVER["HTTP_CONNECTION"]	keep-alive
_SERVER["HTTP_CACHE_CONTROL"]	max-age=0
_SERVER["HTTP_ACCEPT"]	text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
_SERVER["HTTP_UPGRADE_INSECURE_REQUESTS"]	1
_SERVER["HTTP_USER_AGENT"]	Mozilla/5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.82 Safari/537.36
_SERVER["HTTP_ACCEPT_ENCODING"]	gzip, deflate, sdch
_SERVER["HTTP_ACCEPT_LANGUAGE"]	en-US,en;q=0.8
_SERVER["HTTP_COOKIE"]	testing=1; sid=ac430136e1e63a636308f3960eb05833; PHPSESSID=asdfghtle0bf42sgf8326485f3
_SERVER["PATH"]	/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin
_SERVER["SERVER_SIGNATURE"]	no value
_SERVER["SERVER_SOFTWARE"]	Apache/2.4.6 (CentOS) OpenSSL/1.0.1e-fips PHP/5.4.16
_SERVER["SERVER_NAME"]	123.123.123.123
_SERVER["SERVER_ADDR"]	172.20.5.75
_SERVER["SERVER_PORT"]	80
_SERVER["REMOTE_ADDR"]	11.11.11.11
_SERVER["DOCUMENT_ROOT"]	/var/www/html/example.com/html
_SERVER["REQUEST_SCHEME"]	http
_SERVER["CONTEXT_PREFIX"]	no value
_SERVER["CONTEXT_DOCUMENT_ROOT"]	/var/www/html/example.com/html
_SERVER["SERVER_ADMIN"]	root@localhost
_SERVER["SCRIPT_FILENAME"]	/var/www/html/example.com/html/phpinfo.php
_SERVER["REMOTE_PORT"]	52028
_SERVER["GATEWAY_INTERFACE"]	CGI/1.1
_SERVER["SERVER_PROTOCOL"]	HTTP/1.1
_SERVER["REQUEST_METHOD"]	GET
_SERVER["QUERY_STRING"]	no value
_SERVER["REQUEST_URI"]	/phpinfo.php
_SERVER["SCRIPT_NAME"]	/phpinfo.php
_SERVER["PHP_SELF"]	/phpinfo.php
_SERVER["REQUEST_TIME_FLOAT"]	1453653457.192
_SERVER["REQUEST_TIME"]	1453653457

```

**Comment by KyferEz** [comment]
I did some more testing and have found that the variables it's checking for are empty:
PMA_getenv('HTTP_FRONT_END_HTTPS')
PMA_getenv('HTTP_X_FORWARDED_PROTO')
PMA_getenv('HTTPS')
PMA_getenv('HTTP_HTTPS_FROM_LB')

So does this mean something isn't right on my server? Where do I find the PMA_getenv code to look at it?

**Comment by nijel** [comment]
Same issue as #11412. Set PmaAbsoluteUri to workaround that.

**Comment by KyferEz** [comment]
Using PmaAbsoluteUri results in Server 500 error.

**Comment by nijel** [comment]
Anything in the logs?

**Comment by KyferEz** [comment]
No, nothing appears in either /etc/httpd/logs/error_log or  /var/www/html/example.com/error.log

My PmaAbsoluteUri is set as follows because I do not yet have a domain name associated to the server:
$cfg['PmaAbsoluteUri'] = 'https://123.123.123.123/phpadm/'

**Comment by KyferEz** [comment]
LOL I realized I left the ; off the end of the PmaAbsoluteUri line.

In any event, even with the PmaAbsoluteUri line now entered I am still getting a Redirect Loop AND there is nothing in the logs. 

Ideas?

**Comment by KyferEz** [comment]
You know, this does Not make any sense. Webmin works perfectly to halt a http page site when https is required. That works on my server properly. I think your code needs further review and comparison against what Webmin does in order to detect HTTP/HTTPS properly.

**Comment by nijel** [comment]
Actually this code needs to be mostly dropped. It's there really for ancient browsers which we do not support anyway... See https://github.com/phpmyadmin/phpmyadmin/issues/11412 for more details.

**Comment by KyferEz** [comment]
Well how do you recommend I make phpMyAdmin stop loading on port 80 and only use SSL? Even with the PmaAbsoluteUri set it still loads in HTTP if I set ForceSSL=false. If I set it to true I continue to get redirect loops. :(

**Comment by nijel** [comment]
Yes, that's probably best workaround for now. Once we settle down current issues with releases and security fixes, I'll fix work on this (#11412).
